### ISSUE TITLE
the ability to autocomplete the schema not only with tables, but also with functions and procedures

### ISSUE THREAD

**Issue by detkliff** [dbeaver/dbeaver#26839]
Debeaver 24.0.3
Postgresql 14.3

Current version of Debeaver autocomplete tables and view only.
![image](https://github.com/dbeaver/dbeaver/assets/61137576/ec29bd43-8823-4a5e-900d-f143686b38dc)

It would be very convenient to display a list of procedures and functions of the circuit in autocompletion. For example, the last block.
How does Oracle PlSQLDeveloper do this?

**Comment by E1izabeth** [comment]
DBeaver shows completion proposals for procedures and functions in SELECT and WHERE clauses, if this setting is turned on 
![image](https://github.com/dbeaver/dbeaver/assets/28875055/7b63a01b-e7f4-4d7b-aa26-8dabc56004de)

Do you need a procedure in FROM clause?

**Comment by detkliff** [comment]
> Do you need a procedure in FROM clause?

yes, i need.

**Comment by detkliff** [comment]
> DBeaver shows completion proposals for procedures and functions in SELECT and WHERE clauses, if this setting is turned on ![image](https://private-user-images.githubusercontent.com/28875055/327454723-7b63a01b-e7f4-4d7b-aa26-8dabc56004de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2NjQ5MjcsIm5iZiI6MTcxNDY2NDYyNywicGF0aCI6Ii8yODg3NTA1NS8zMjc0NTQ3MjMtN2I2M2EwMWItZTdmNC00ZDdiLWFhMjYtOGRhYmM1NjAwNGRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAyVDE1NDM0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjNDBmNDk0N2MyMDQ0NWY5MGQwZDJlNGZhOTkzZDIyODZkZTY0MGQ5ZTAyMmUwM2NiN2VjODY3MjZmMzcyOTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wu-QX2UZuz5BKPBu-F_CLzKGnrmCaofD-r5AD-kgcys)
> 
> Do you need a procedure in FROM clause?

but your option works not well
step 1 wrong search
![image](https://github.com/dbeaver/dbeaver/assets/61137576/001924f3-b1c9-4cbd-b939-f2ce11b6d2bd)
step 2 wrong search
![image](https://github.com/dbeaver/dbeaver/assets/61137576/0a050b0e-229a-4221-9044-f99714bc2909)
step 3 ok
![image](https://github.com/dbeaver/dbeaver/assets/61137576/ef8133a4-6f58-4983-bb18-e21851e66fee)
step 4 ok
![image](https://github.com/dbeaver/dbeaver/assets/61137576/1d58056d-56ac-45de-9184-15aba8e1f531)

**Comment by E1izabeth** [comment]
>but your option works not well

It doesn't work with schema name, but works without it.
![image](https://github.com/dbeaver/dbeaver/assets/28875055/827a045a-0dee-48a5-af70-5eeb3fbc35a3)

**Comment by detkliff** [comment]
> > but your option works not well
> 
> It doesn't work with schema name, but works without it. ![image](https://private-user-images.githubusercontent.com/28875055/327497777-827a045a-0dee-48a5-af70-5eeb3fbc35a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2NjY5MTYsIm5iZiI6MTcxNDY2NjYxNiwicGF0aCI6Ii8yODg3NTA1NS8zMjc0OTc3NzctODI3YTA0NWEtMGRlZS00OGE1LWFmNzAtNWVlYjNmYmMzNWEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAyVDE2MTY1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwNGZjNjY5NjM2NTJkYzU1M2I4NDEzYzJhMjgwOWI3Y2Q2ZmYzZmQwNGFiNWE1MDc4OTE1YTAxYWVlMWJhOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.x11bRC9v3spE_6IxQ8ZxVkJA0JFG6JHOkiShO3ZYJEQ)

It’s not at all clear which function it will call if there are functions with the same names and signatures in several schemas

**Comment by E1izabeth** [comment]
Yes, I agree with you. That's a point to improve autocompletion too.

**Comment by detkliff** [comment]
I had to disable this option. I don't understand how it works at all.
![image](https://github.com/dbeaver/dbeaver/assets/61137576/8f48e348-32a7-42fc-ac1b-bb88f5445000)

After disabling the option, it no longer works correctly

![image](https://github.com/dbeaver/dbeaver/assets/61137576/f54bed03-0c09-4a13-bb91-4d10cafc67d6)
but I haven't specified the point after the table alias yet

if I choose the suggested option, the alias is replaced with the column name and the alias disappears
![image](https://github.com/dbeaver/dbeaver/assets/61137576/0bfe51e5-8310-49fa-a4d4-b41547bd1957)

**Comment by detkliff** [comment]
![image](https://github.com/dbeaver/dbeaver/assets/61137576/640a98fc-f35f-4103-a458-723e75bd823d)
what kind of list this is is not clear, it does not apply to either nsi_io or nsi_candidate

**Comment by E1izabeth** [comment]
Completion doesn't work well in subqueries now. We're working on improvements.
